components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -91,11 +91,15 @@ void RewardsNotificationServiceImpl::DeleteNotification(
     rewards_notification = rewards_notifications_[id];
     rewards_notifications_.erase(id);
   }
+  StoreRewardsNotifications();
```

Comment:
why this change?

Reply:
Exactly the same problem as above, we only store on destructor which is wrong.

Reply:
I restored it in destructor. @NejcZdovc  @SergeyZhukovsky is it enough to resolve this?

Reply:
either way we specifically don't want to add stuff like this here because it makes unit testing a nightmare. That is what the observer methods are for. If you need to do something when a notification is deleted then add an observer for `OnNotificationDeleted`

Reply:
@bridiver it's not to do something, desktop stores notifications in a file on destructor of that class only. It doesn't work for Android as never browser exits in normal mode. We need to refresh the notification storage on every delete and add

Reply:
@SergeyZhukovsky I think I misunderstood what this was doing on the first pass through and thought it was something else, but separately @tmancey and @NejcZdovc why are we serializing this to json for `kRewardsNotifications` when we can just directly write a dictionary value or a list value?

Reply:
```
std::string result;
  if (!base::JSONWriter::Write(root, &result)) {
    LOG(ERROR) << "Failed to serialize rewards notifications on shutdown";
    return;
  }

  profile_->GetPrefs()->SetString(prefs::kRewardsNotifications, result);
```
you can call `GetPrefs()->Set(prefs::kRewardsNotifications, root);`

Reply:
@bridiver as far as I understand this change is not part of this PR. Should we create separate issue for this?